sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/interop/LLVMForeignCallNode.java

```diff
@@ -188,19 +190,20 @@ public static PackForeignArgumentsNode create(FunctionType bitcodeFunctionType,
     }
 
     @CompilationFinal private ContextReference<LLVMContext> ctxRef;
-    private final LLVMInteropType.Structured returnBaseType;
+    protected final LLVMInteropType.Structured returnBaseType;
 
     @Child LLVMGetStackFromThreadNode getStack;
     @Child DirectCallNode callNode;
     @Child LLVMDataEscapeNode prepareValueForEscape;
     @Child PackForeignArgumentsNode packArguments;
 
-    public LLVMForeignCallNode(LLVMLanguage language, LLVMFunctionDescriptor function, LLVMInteropType interopType, LLVMSourceFunctionType sourceType) {
+    protected LLVMForeignCallNode(LLVMLanguage language, LLVMFunctionDescriptor function, LLVMInteropType interopType, LLVMSourceFunctionType sourceType, Structured returnBaseType, Type escapeType) {
         super(language);
-        this.returnBaseType = getReturnBaseType(interopType);
+        this.returnBaseType = returnBaseType;
         this.getStack = LLVMGetStackFromThreadNode.create();
         this.callNode = DirectCallNode.create(getCallTarget(function));
-        this.prepareValueForEscape = LLVMDataEscapeNode.create(function.getLLVMFunction().getType().getReturnType());
+        this.callNode.forceInlining();
```

Comment:
I think this is a wrong merge conflicting with e8b0489b4e69108949049e887145d45293eff0e5, which removed the `forceInlining`.

Reply:
Indeed, thanks!